From 3cafb325abe86e023a46a89ad1d2db6a20c0e586 Mon Sep 17 00:00:00 2001
From: meixie <meixie@qti.qualcomm.com>
Date: Thu, 23 Jan 2025 16:46:38 +0800
Subject: [PATCH 08/11] ARM: dts: msm: Enable emac pass thru

Define eth driver device tree:
lemans-vm-la-mt.dtsi is used for emac1 pass thru to LA GVM.

Change-Id: I898c5847aca2330d4dac7a07ccea5b26fb804487
Signed-off-by: meixie <quic_meixie@quicinc.com>
Signed-off-by: Samuel Lee <samuelee@qti.qualcomm.com>
---
 qcom/lemans-vm-la-mt.dtsi | 109 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 109 insertions(+)

diff --git a/qcom/lemans-vm-la-mt.dtsi b/qcom/lemans-vm-la-mt.dtsi
index 7b5eb19dc..fcf761fb3 100644
--- a/qcom/lemans-vm-la-mt.dtsi
+++ b/qcom/lemans-vm-la-mt.dtsi
@@ -89,6 +89,115 @@
 			pcm-lane-config = <1>;
 		};
 	};
+
+	emac1_mtl_rx_setup: rx-queues-config {
+		snps,rx-queues-to-use = <4>;
+		snps,rx-sched-sp;
+
+		queue0 {
+			snps,dcb-algorithm;
+			snps,map-to-dma-channel = <0x0>;
+			snps,route-up;
+			snps,priority = <0x1>;
+		};
+
+		queue1 {
+			snps,dcb-algorithm;
+			snps,map-to-dma-channel = <0x1>;
+			snps,route-ptp;
+		};
+
+		queue2 {
+			snps,avb-algorithm;
+			snps,map-to-dma-channel = <0x2>;
+			snps,route-avcp;
+		};
+
+		queue3 {
+			snps,avb-algorithm;
+			snps,map-to-dma-channel = <0x3>;
+			snps,priority = <0xC>;
+		};
+	};
+
+	emac1_mtl_tx_setup: tx-queues-config {
+		snps,tx-queues-to-use = <4>;
+		snps,tx-sched-sp;
+		queue0 {
+			snps,dcb-algorithm;
+		};
+
+		queue1 {
+			snps,dcb-algorithm;
+		};
+
+		queue2 {
+			snps,avb-algorithm;
+			snps,send_slope = <0x1000>;
+			snps,idle_slope = <0x1000>;
+			snps,high_credit = <0x3E800>;
+			snps,low_credit = <0xFFC18000>;
+		};
+
+		queue3 {
+			snps,avb-algorithm;
+			snps,send_slope = <0x1000>;
+			snps,idle_slope = <0x1000>;
+			snps,high_credit = <0x3E800>;
+			snps,low_credit = <0xFFC18000>;
+		};
+
+	};
+
+
+
+	ethqos_hw1: qcom,ethernet@23000000 {
+		compatible = "qcom,stmmac-ethqos","snps,dwmac-4.20a";
+		qcom,arm-smmu;
+		reg = <0x23000000 0x10000>,
+				<0x23016000 0x100>,
+				<0x08902000 0xE10>;
+		reg-names = "stmmaceth","rgmii","serdes";
+		clocks = <&gcc GCC_EMAC1_AXI_CLK>,
+				<&gcc GCC_EMAC1_SLV_AHB_CLK>,
+				<&gcc GCC_EMAC1_PTP_CLK>,
+				<&gcc GCC_EMAC1_PHY_AUX_CLK>,
+				<&gcc GCC_SGMI_CLKREF_EN>,
+				<&gcc GCC_EMAC1_RGMII_CLK>;
+		clock-names = "stmmaceth", "pclk", "ptp_ref", "phyaux","sgmi_ref","rgmii";
+
+		snps,ptp-ref-clk-rate = <230400000>;
+		snps,ptp-req-clk-rate = <125000000>;
+		interrupts-extended = <&intc 0 929 4>;
+		interrupt-names = "macirq";
+
+		snps,tso;
+		snps,pbl = <32>;
+		mac-address = [AA BB CC DD 00 02];
+		rx-fifo-depth = <32768>;
+		tx-fifo-depth = <32768>;
+		snps,mtl-rx-config = <&emac1_mtl_rx_setup>;
+		snps,mtl-tx-config = <&emac1_mtl_tx_setup>;
+		gdsc_emac-supply = <&gcc_emac1_gdsc>;
+		vreg_emac_phy-supply = <&pm8775_a_l5>;
+		snps,ps-speed = <1000>;
+		phy-mode = "sgmii";
+		pcs-v3;
+
+		fixed-link {
+			status = "okay";
+			speed = <2500>;
+			full-duplex;
+		};
+
+		ethqos_emb_smmu: ethqos_emb_smmu {
+			compatible = "qcom,emac-smmu-embedded";
+			iommus = <&apps_smmu 0x140 0xf>;
+			qcom,iommu-dma = "fastmap";
+			qcom,iommu-dma-addr-pool = <0x80000000 0x40000000>;
+		};
+	};
+
 };
 
 / {
-- 
2.34.1

